lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,124 @@
+var edx = edx || {},
+
+    Dropdown = (function() {
+
+        var _dropdown = {
+
+            opts: {
+                page: $(document),
+                midpoint: ($(window).width() / 2),
+                button: $('.button-more.has-dropdown'),
+                button_active: 'is-active',
+                menu: $('.dropdown'),
+                menu_inactive: 'is-hidden',
+                menu_active: 'is-visible',
+                menu_align: 'align-',
+                text_align: 'control-text-align'
+            },
+
+            init: function() {
+                _dropdown.listenForClick();
+                _dropdown.listenForKeypress();
+            },
+
+            listenForClick: function() {
+                var that = _dropdown;
+
+                that.opts.button.on('click', function() {
+                    that.closeDropdownMenu();
+                    that.openDropdownMenu($(this));
+                });
+
+                that.opts.page.on('click', function() {
+                    that.closeDropdownMenu();
+                });
+            },
+
+            listenForKeypress: function() {
+                var that = _dropdown;
+
+                that.opts.page.keyup(function(e) {
+                    if (27 === e.which) {
+                        that.closeDropdownMenus();
+                    }
+                });
+            },
+
+            closeDropdownMenu: function() {
+                var that = _dropdown;
+                    open = that.opts.page.find(that.opts.menu).not(':focus');
+
+                if (open) {
+                    open
+                        .removeClass(that.opts.menu_active)
+                        .addClass(that.opts.menu_inactive);
+
+                    open.parent()
+                        .find(that.opts.button)
+                            .removeClass(that.opts.button_active)
```

Comment:
nitpick: can we move the indentation so that .find, .removeClass and .attr all have the same indentation?